helm/cas-airflow-dag-trigger/templates/dag-trigger-job.yaml

```diff
@@ -0,0 +1,61 @@
+{{- if .Values.enable }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "cas-airflow-dag-trigger.fullname" . }}
+  labels:
+{{ include "cas-airflow-dag-trigger.labels" . | indent 4 }}
+{{- if .Release.IsUpgrade }}
+# This job requires the cas-postgres dependency to be running,
+# and is required for the app deployment to become ready.
+# Therefore, it needs to be part of the initial install (instead of a pre/post install hook)
+# For upgrades, it needs to be a hook so that the delete policy kicks in
+# and the previous job template gets deleted (as editing job templates is not allowed)
+# It can be a hook during upgrades as the cas-postgres dependency will already be running
+  annotations:
+    "helm.sh/hook": "pre-upgrade"
+    "helm.sh/hook-delete-policy": "before-hook-creation"
+    "helm.sh/hook-weight": "0"
+{{- end }}
```

Comment:
We should have a set of values in the chart to control when the job is run.

`helmHook` and `helmHookWeight` are two values that come to mind (`before-hook-creation` is the default delete policy and probably doesn't need to even be in the template).
The comments above are not quite accurate. The job does not have to be a hook during upgrades, all we need is to add a random suffix to the job name so that helm deletes the old job and creates a new one, instead of trying to update the job. 
See https://github.com/bcgov/cas-ggircs/blob/81bdaeb45e384911254c4a4a50efc92bfd877916/helm/cas-ggircs/templates/etl/airflow-dag-trigger.yaml#L5

I would only add this random suffix if the job is not run as a hook (i.e. `helmHook` is `null`).

We do have the need for both use cases:
 - the DAG that pulls the other repo's DAGs will need to be run as a `pre` hook
 - the deployment DAGs will be triggered during the actual deployment of the helm charts